src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -541,6 +567,21 @@ public static void WriteLengthPrefixedString(BinaryWriter writer, string value)
             writer.Write(value.ToCharArray());
         }
 
+        /// <summary>
+        /// Reads the value of <see cref="CommitHashAttribute.Hash"/> of the assembly <see cref="BuildRequest"/> is defined in
+        /// </summary>
+        /// <returns>The hash value of the current assembly or an empty string</returns>
+        public static string GetCommitHash()
+        {
+            var hashAttribute = typeof(BuildRequest).Assembly.GetCustomAttribute<CommitHashAttribute>();
```

Comment:
A bit of an edge case but `GetCustomAttribute` can still throw here if we truly mess up and apply `CommitHashAttribute` twice. Should probably guard against that.